Makefile.build

```diff
@@ -247,8 +247,8 @@ config-sync-manifest: config-sync-manifest-no-push push-images
 # config-sync-manifest-local builds config sync for local testing in kind.
 # starts local docker registry and pushes images to the local registry
 .PHONY: config-sync-manifest-local
-config-sync-manifest-local:
-	@bash scripts/docker-registry.sh
+config-sync-manifest-local: install-kind
```

Comment:
i use this target all the time locally. do i really want it to install kind for CI?

Reply:
config-sync-manifest-local-ci could depend on install-kind-ci, while config-sync-manifest-local could depend on install-kind.

Reply:
or... is there a way to tell if we're in CI with an env var or something and use that to decide where to install kind?

Reply:
Main goal with this is to install kind in the CI job at runtime, so that it's possible to change the kind version and have presubmits pass.

With the preinstalled kind version, we basically have to:
- publish a new image with a bumped kind version
- update CI to use the new image, which will break our presubmits until we bump the kind version used in the repo
- create a PR that bumps the kind version, and submit to fix presubmits

This requires putting the CI in a broken state in the interim of an upgrade, which is undesirable. Using the dynamic kind version like this makes the upgrade look like:
- create a PR that bumps the kind version

That being said, I could move this as a dependency of a different target so that it isn't called every time you run `config-sync-manifest-local` locally.

Reply:
Moved the dependency to another make target (the e2e-kind entrypoint used by CI)

Reply:
I don't think there's a need to treat CI/local differently in this case